azure-pipelines.yml

```diff
@@ -35,39 +35,49 @@ jobs:
     pool:
       vmImage: 'macOS-10.14'
     variables:
-      cache-key: $(Build.StagingDirectory)/macos-nix-key
-      cache-path: /tmp/cache/
+      nix-cache-key: $(Build.StagingDirectory)/macos-nix-key
+      nix-cache-path: /tmp/nix-cache/
+      bazel-repo-cache-key: $(Build.StagingDirectory)/bazel-repo-cache-key
+      bazel-repo-cache-path: $(Agent.BuildDirectory)/.bazel-cache/repo
     steps:
       - template: ci/report-start.yml
       - checkout: self
-      - bash: echo $(git log -n1 --pretty=format:%H dev-env nix azure-pipelines.yml) >> $(cache-key)
-        displayName: cache key
+      - bash: echo $(git log -n1 --pretty=format:%H dev-env nix azure-pipelines.yml) >> $(nix-cache-key)
+        displayName: nix cache key
       - task: CacheBeta@0
         inputs:
-          key: $(cache-key)
-          path: $(cache-path)
+          key: $(nix-cache-key)
+          path: $(nix-cache-path)
       - bash: |
           set -euo pipefail
-          if [[ -e $(cache-path) ]]; then
+          if [[ -e $(nix-cache-path) ]]; then
               DIR=$(pwd)
               sudo mkdir /nix && sudo chown $USER /nix
               cd /nix
-              tar xzf $(cache-path)/nix.tar.gz
+              tar xzf $(nix-cache-path)/nix.tar.gz
               cd $DIR
               curl -sfL https://nixos.org/releases/nix/nix-2.2.1/install | bash
           fi
         displayName: restore cache
+      - bash: echo $(git log -n1 --pretty=format:%H azure-pipelines.yml $(find . -name \*.bazel -or -name \*.bzl -or -name WORKSPACE -or -name BUILD)) >> $(bazel-repo-cache-key)
```

Comment:
As I understand the [`--repository_cache` flag](https://bazel.build/designs/2016/09/30/repository-cache.html) this only caches external dependencies, i.e. dependencies specified in `WORKSPACE` or files imported by `WORKSPACE`. So, I don't think `*.bazel` or `BUILD` need to be included in the key. @cocreature am I missing something?